gateway-js/src/index.ts

```diff
@@ -424,11 +435,8 @@ export class ApolloGateway implements GraphQLService {
       )
     } else {
       this.schema = schema;
-      if (this.queryPlannerPointer != null) {
-        // Allow rust to release the memory for this QueryPlanner
-        dropQueryPlanner(this.queryPlannerPointer)
-      }
       this.queryPlannerPointer = getQueryPlanner(composedSdl);
```

Comment:
So previously when we updated the schema for a particular gateway object, we'd also drop the old query planner. But now when we update the schema, we'll keep the old query planners until `cleanup` is called.  Is that right?

In webapp we have two gateway subclasses - I think this case is fine for the VersionedGateway case, which does not update its schema and eventually calls `cleanup` when its evicted from cache, but I think we could run into potential memory issues for the DefaultVersionedGateway case, which is long lived and doesn't ever really call cleanup.

I'm not sure what the best way to handle that is - maybe instead of a single 'cleaned_up' bool, we keep a map of query planner pointer -> num requests in use, then within `trackQueryPlanUse` we drop pointers that no longer have requests in use? I suppose we'd also need some list of `deprecatedQueryPlanPointers` in place of / in addition to `ownedQueryPlannerPointers`, since we don't want to drop the 'main' query plan pointer if there doesn't happen to be any in flight requests.

Reply:
hmmmmm ok I was hoping to avoid doing a full reference counting setup, but maybe that's unavoidable 😅